db/db.go

```diff
@@ -181,6 +186,9 @@ func (b *boltDB) Stop(_ context.Context) error {
 
 // Put inserts a <key, value> record
 func (b *boltDB) Put(namespace string, key, value []byte) error {
+	b.mutex.Lock()
```

Comment:
Do not quite follow why Put doesn't have a lock. Let me know if this is on purpose.

Reply:
inside (db *DB) Update(), it calls db.Begin(true) --> db.beginRWTx() or db.beginTx(), inside there there is lock to protect the Update() as a transaction, so my understanding is that we don't need a lock here